src/app/jscaip/tests/MCTS.spec.ts

```diff
@@ -0,0 +1,75 @@
+/* eslint-disable max-lines-per-function */
+import { AwaleMove } from 'src/app/games/awale/AwaleMove';
+import { AwaleMoveGenerator } from 'src/app/games/awale/AwaleMoveGenerator';
+import { AwaleRules } from 'src/app/games/awale/AwaleRules';
+import { AwaleState } from 'src/app/games/awale/AwaleState';
+import { QuartoMove } from 'src/app/games/quarto/QuartoMove';
+import { QuartoMoveGenerator } from 'src/app/games/quarto/QuartoMoveGenerator';
+import { QuartoPiece } from 'src/app/games/quarto/QuartoPiece';
+import { QuartoNode, QuartoRules } from 'src/app/games/quarto/QuartoRules';
+import { QuartoState } from 'src/app/games/quarto/QuartoState';
+import { Table } from 'src/app/utils/ArrayUtils';
+import { AITimeLimitOptions } from '../AI';
+import { Coord } from '../Coord';
+import { MCTS } from '../MCTS';
+
+describe('MCTS', () => {
+
+    let mcts: MCTS<QuartoMove, QuartoState>;
+    const mctsOptions: AITimeLimitOptions = { name: '100ms', maxSeconds: 0.1 };
+
+    beforeEach(() => {
+        mcts = new MCTS('MCTS', new QuartoMoveGenerator(), QuartoRules.get());
+    });
+    it('should not be stupid', () => {
+        // Given a board that could be a win for opponent in their next moves
+        const board: Table<QuartoPiece> = [
+            [QuartoPiece.AAAA, QuartoPiece.AAAB, QuartoPiece.AABB, QuartoPiece.EMPTY],
+            [QuartoPiece.EMPTY, QuartoPiece.EMPTY, QuartoPiece.EMPTY, QuartoPiece.EMPTY],
+            [QuartoPiece.EMPTY, QuartoPiece.EMPTY, QuartoPiece.EMPTY, QuartoPiece.EMPTY],
+            [QuartoPiece.EMPTY, QuartoPiece.EMPTY, QuartoPiece.EMPTY, QuartoPiece.EMPTY],
+        ];
+        const state: QuartoState = new QuartoState(board, 14, QuartoPiece.BBAA);
+        const node: QuartoNode = new QuartoNode(state);
+
+        // When computing the best move
+        const move: QuartoMove = mcts.chooseNextMove(node, mctsOptions);
+        // Then it should not give the win to the opponent
+        if (move.coord.equals(new Coord(3, 0))) {
+            // MCTS blocked the opponent's only possible win
+            expect(true).toBe(true);
+        } else {
+            // MCTS gave the opponent a piece with which they cannot win
+            expect(move.piece).not.toBe(QuartoPiece.AABA);
+        }
+    });
```

Comment:
![Capture d’écran 2023-08-18 111324](https://github.com/EveryBoard/EveryBoard/assets/41200522/370540d8-5991-46d1-b42f-392c076a3a55)
J'ai peur que ton test ne soit qu'une chance, elle donne la pièce gagnante !

Reply:
c'est pas une chance, le test est construit pour qu'on veuille sélectionner ce move (car statistiquement on aura exploré assez de parties pour que ça soit avantageux), alors que dans la screenshot le hasard fait que statistiquement, donner la pièce de la défaite est pas désavantageux (même si c'est con)

Reply:
donc l'intitulé c'est should not be stupid, mais il est stupide quand même :D !
MCTS me déçoit pour l'instant :D !
J'suis content cependant qu'alpha beta ai un adversaire pour prouver sa supériorité :D !

Reply:
Ok donc je renomme le it en :
```
    it('should choose possible victory over definite defeat', () => {
```